gsy_framework/schema/validators.py

```diff
@@ -49,6 +56,128 @@ def validate(self, data, raise_exception: bool = False) -> (bool, str):
+        return data
+
+    def modify_output(self, data: Dict) -> Dict:
+        """Modify the output data dict in order for AVRO  to properly deserialize it."""
+        return data
+
+    def serialize(self, data: Dict, raise_exception: bool) -> bytes:
+        """Validate and serialize dictionary."""
+        data = self.modify_input(data)
+        is_valid, errors = self.validate(data, raise_exception)
+        if not is_valid:
+            raise GSySerializationException(f"Failed to serialize data: {errors}")
+        writer = avro.io.DatumWriter(self.schema)
+        bytes_writer = BytesIO()
+        encoder = avro.io.BinaryEncoder(bytes_writer)
+        writer.write(data, encoder)
+        return bytes_writer.getvalue()
+
+    def deserialize(self, data: bytes) -> Dict:
+        """Deserialize and validate dictionary."""
+        try:
+            reader = avro.io.DatumReader(self.schema)
+            bytes_reader = BytesIO(data)
+            decoder = avro.io.BinaryDecoder(bytes_reader)
+            deserialized_dict = reader.read(decoder)
+        except AvroException as ex:
+            raise GSySerializationException(f"Failed to deserialize data: {str(ex)}.") from ex
+        deserialized_dict = self.modify_output(deserialized_dict)
+        return deserialized_dict
+
+
+class AVROSimulationSettingsSerializer(AVROSchemaSerializer):
+    """Serializer class for the simulation settings."""
+
+    def __init__(self):
+        super().__init__("launch_simulation_settings")
+
+    def modify_input(self, data: Dict) -> Dict:
+        try:
+            data["duration"] = int(data["duration"].total_seconds())
+            data["slot_length"] = int(data["slot_length"].total_seconds())
+            data["tick_length"] = int(data["tick_length"].total_seconds())
+            data["slot_length_realtime"] = int(data["slot_length_realtime"].total_seconds())
+        except (KeyError, ValueError, TypeError) as ex:
+            error_log = (
+                "Time related parameters on the settings were not set or incorrectly assigned.")
+            logger.error(error_log)
+            raise GSySerializationException(error_log) from ex
+        return data
+
+    def modify_output(self, data: Dict) -> Dict:
+        try:
+            data["duration"] = timedelta(seconds=data["duration"])
+            data["slot_length"] = timedelta(seconds=data["slot_length"])
+            data["tick_length"] = timedelta(seconds=data["tick_length"])
+            data["slot_length_realtime"] = timedelta(seconds=data["slot_length_realtime"])
+        except (KeyError, ValueError, TypeError) as ex:
+            error_log = (
+                "Time related parameters on the settings were not set or incorrectly assigned.")
+            logger.error(error_log)
+            raise GSySerializationException(error_log) from ex
+        return data
+
+
 def get_schema_validator(schema_name: str) -> BaseSchemaValidator:
```

Comment:
I suggest to update and stick with this function in order to abstract AVRO from the using code.

Reply:
Will do so, thanks